src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -393,13 +394,35 @@ public Result getResultForParticipationAndCheckAccess(Long participationId, Long
     private List<Feedback> saveFeedbackWithHibernateWorkaround(@NotNull Result result, List<Feedback> feedbackList) {
         // Avoid hibernate exception
         List<Feedback> savedFeedbacks = new ArrayList<>();
+        // Collect ids of feedbacks that have long feedback.
+        List<Long> feedbackIdsWithLongFeedback = feedbackList.stream().filter(feedback -> feedback.getId() != null && feedback.getHasLongFeedbackText()).map(Feedback::getId)
+                .toList();
```

Comment:
To make absolutely sure that no unnecessary DB calls are made, you could maybe additionally check if the feedbacks actually have long feedback texts, or if the list is empty.
```suggestion
        // Collect ids of feedbacks that have long feedback.
        List<Long> feedbackIdsWithMissingLongFeedback = feedbackList.stream().filter(feedback -> feedback.getId() != null && feedback.getHasLongFeedbackText() && feedback.getLongFeedback().isEmpty()).map(Feedback::getId)
                .toList();
```
However, if we do this, some parent feedbacks might keep their association to the long feedback texts (lines 409-416). The question is if we really need to remove the associations. 

An alternative would be to create associations for the missing ones, so that we don't have to manually re-fetch them on every save. But I think both approaches are valid and solve the problem.